src/torchmetrics/metric.py

```diff
@@ -858,6 +858,9 @@ def __getitem__(self, idx: int) -> "Metric":
     def __getnewargs__(self) -> Tuple:
         return (Metric.__str__(self),)
 
+    def __iter__(self):
```

Comment:
this dies not apply to metric collection right?

Reply:
No, because metric collection is actually not a metric:
https://github.com/Lightning-AI/metrics/blob/a3dc40c89e79622474290e8813ff27b92ffd8268/src/torchmetrics/collections.py#L29